docs/api/rtos/Mutex.md

```diff
@@ -1,10 +1,24 @@
 ## Mutex
 
-<span class="images">![](https://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/classrtos_1_1_mutex.png)<span>Mutex class hierarchy</span></span>
+### Mutex class hierarchy
+
+<span class="images">![](https://os-doc-builder.test.mbed.com/docs/development/mbed-os-api-doxy/classrtos_1_1_mutex.png)</span>
 
 A Mutex is used to synchronize the execution of threads, for example to protect the access to a shared resource.
 
-<span class="notes"> **Note:** The `Mutex` methods cannot be called from interrupt service routines (ISR). In the current version of Mbed OS, if you attempt to use a mutex from within an ISR, nothing happens; attempts to lock a mutex succeed immediately, regardless of whether the lock is actually free. In other words, if you acquire a mutex lock in an ISR, you can break the thread safety mechanisms and introduce race-conditions into an otherwise safe piece of code. Future versions of Mbed OS will provide warnings and ultimately prevent this from happening. </span>
+<span class="notes"> **Note:** The `Mutex` methods cannot be called from interrupt service routines (ISR). In the current version of Mbed OS, if you attempt to use a mutex from within an ISR, you see an error like this: </span>
```

Comment:
Can we change
`if you attempt to use a mutex from within an ISR, you see an error like this:`

to

`if you attempt to use a mutex from within an ISR, it will treat it as fatal system error and you see an error like this`

Reply:
Also we can add if synchronization is required in ISR, consider using semaphores

Reply:
Both good points. @AnotherButler can we hold off the merge of this PR while I work on these changes.?

Reply:
"if you attempt to use a mutex from within an ISR, the ISR treats the mutex as a fatal system error, and you see an error like this:"?